contracts/JBETHPaymentTerminal.sol

```diff
@@ -3,926 +3,54 @@
-
-    if (_balance > 0)
-      // Withdraw the balance to transfer to the new terminal;
-      _to.addToBalanceOf{value: _balance}(_projectId, '');
-
-    emit Migrate(_projectId, _to, _balance, msg.sender);
-  }
-
-  /**
-    @notice
-    Receives funds belonging to the specified project.
-
-    @param _projectId The ID of the project to which the funds received belong.
-    @param _memo A memo to pass along to the emitted event.
-  */
-  function addToBalanceOf(uint256 _projectId, string memory _memo)
-    external
-    payable
-    override
-    nonReentrant
-    isTerminalOfProject(_projectId)
-  {
-    // Amount must be greater than 0.
-    if (msg.value == 0) {
-      revert ZERO_VALUE_SENT();
-    }
-
-    // Record the added funds.
-    store.recordAddedBalanceFor(_projectId, msg.value);
-
-    // Refund any held fees to make sure the project doesn't pay double for funds going in and out of the protocol.
-    _refundHeldFees(_projectId, msg.value);
-
-    emit AddToBalance(_projectId, msg.value, _memo, msg.sender);
-  }
-
-  /**
-    @notice
-    Process any fees that are being held for the project.
-
-    @dev
-    Only a project owner, an operator, or the contract's owner can process held fees.
-
-    @param _projectId The ID of the project whos held fees should be processed.
-  */
-  function processFees(uint256 _projectId)
-    external
-    override
-    requirePermissionAllowingOverride(
-      projects.ownerOf(_projectId),
-      _projectId,
-      JBOperations.PROCESS_FEES,
-      msg.sender == owner()
+    JBPaymentTerminal(
+      JBTokens.ETH,
+      JBCurrencies.ETH,
+      _baseWeightCurrency,
+      JBSplitsGroups.ETH_PAYOUT,
+      _operatorStore,
+      _projects,
+      _directory,
+      _splitsStore,
+      _store,
+      _owner
     )
-    nonReentrant
+  // solhint-disable-next-line no-empty-blocks
   {
-    // Get a reference to the project's held fees.
-    JBFee[] memory _heldFees = _heldFeesOf[_projectId];
-
-    // Process each fee.
-    for (uint256 _i = 0; _i < _heldFees.length; _i++)
-      _processFee(
-        _heldFees[_i].amount -
-          PRBMath.mulDiv(
-            _heldFees[_i].amount,
-            JBConstants.MAX_FEE,
-            _heldFees[_i].fee + JBConstants.MAX_FEE
-          ),
-        _heldFees[_i].beneficiary
-      );
 
-    // Delete the held fee's now that they've been processed.
-    delete _heldFeesOf[_projectId];
-
-    emit ProcessFees(_projectId, _heldFees, msg.sender);
   }
 
-  /**
-    @notice
-    Allows the fee to be updated for subsequent funding cycle configurations.
-
-    @dev
-    Only the owner of this contract can change the fee.
-
-    @param _fee The new fee.
-  */
-  function setFee(uint256 _fee) external override onlyOwner {
-    // The max fee is 5%.
-    if (_fee > _FEE_CAP) {
-      revert FEE_TOO_HIGH();
+  function _transferFrom(
+    address _from,
+    address payable _to,
+    uint256 _amount
+  ) internal override {
+    if (_from != address(this)) {
```

Comment:
Since _from is only there for interface implementation, do we really need this test?

Reply:
we don't. it's guarenteed that if Token == ETH then this wont be called with a _from that's not equal this.
good catch.